python/lsst/display/firefly/footprints.py

```diff
@@ -0,0 +1,202 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+
+##
+# @file
+# @brief Support for LSST footprints, specific to Firefly
+
+from copy import deepcopy
+import io
+import tempfile
+
+import numpy as np
+from astropy.io.votable.tree import VOTableFile, Resource, Field, Info
+from astropy.io.votable import from_table
+from astropy.table import Table
+import astropy.units as u
+
+import lsst.afw.geom as afwGeom
+
+
+
+def record_selector(record, selection):
+    """Select records from source catalog
+
+    Parameters:
+    -----------
+    record : `lsst.afw.detect.SourceRecord`
+        record to select
+    selection : `str`
+        'all' to select all records. 'blended parents' to select records with
+        more than zero children. 'deblended children' to select records with
+        non-zero parents. 'isolated' to select records that are not blended,
+        meaning zero parents and zero children.
+        Values to check for sel
+    """
+    nChildren = record.get('deblend_nChild')
+    parentId = record.getParent()
+    if selection == 'all':
+        return True
+    elif selection == 'blended parents':
+        return (nChildren > 0)
+    elif selection == 'deblended children':
+        return (parentId > 0)
+    elif selection == 'isolated':
+        return ((parentId == 0) and (nChildren == 0))
+    else:
+        raise RuntimeError('invalid selection: {}'.format(selection) + 
+                           '\nMust be one of "all", "blended parents", ' +
+                           '"deblended children", "isolated"')
+
+
+def create_footprints_table(catalog, xy0 = afwGeom.Point2I(0,0),
+                            insert_column=6):
+    """make a VOTable of SourceData table and footprints
+
+    Parameters:
+    -----------
+    catalog : `lsst.afw.table.SourceCatalog`
+            Source catalog from which to display footprints.
+    xy0 : `lsst.afw.geom.Point2I`, default afwGeom.Point2I(0,0)
+        Pixel origin to subtract off from the footprint coordinates.
+        Normally this can be left at the default.
+    insert_column : `int`
+        Column at which to insert the "family_id" and "category" columns
+
+    Returns:
+    --------
+    `astropy.io.votable.VOTableFile`
+        VOTable object to upload to Firefly
+    """
+    with tempfile.NamedTemporaryFile() as fd:
+        catalog.writeFits(fd.name)
+        sourceTable = Table.read(fd.name, hdu=1)
```

Comment:
This pattern is going to remove all of the flag information.  Could you use `catalog.asAstropy` instead?

Reply:
After making a continuous copy of the catalog, I can make the `astropy.table.Table` using `catalog.asAstropy()`, but then converting to an `astropy.vo.VOTableFile` using `from_table` fails with an exception `TypeError: dtype('int64') can not be represented in VOTable`.

`sourceTable['id'].dtype` resolves to `dtype('int64')` and `dtype('int64').num` evaluates to `9`.  

Checking for columns with `dtype.num == 9` and replacing the `dtype` with `np.dtype('long')` makes the code work. I don't see 'flags' in the Astropy table returned by `catalog.asAstropy()` though. I am puzzled.